File: Data Structures & Algorithms/validate-parentheses/submission-3.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        count={"}":"{", "]": "[", ")":"("}
        for c in s:
            if c in count:
                if stack and stack[-1] == count[c]:
                    stack.pop()
                else: return False
            else: stack.append(c)
        return True if not stack else False

File: Data Structures & Algorithms/daily-temperatures/submission-3.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int>res(temperatures.size(),0);
        stack<int>stk;

        for(int i=0;i<temperatures.size();i++){
            while(!stk.empty() && temperatures[i]>temperatures[stk.top()]){
                int prev=stk.top();
                stk.pop();
                res[prev]=i-prev;
            }
            stk.push(i);
        }
        return res;
    }
};
